src/interface/idl.rs

```diff
@@ -0,0 +1,594 @@
+    }
+}
+
+/// The [`enum`](EnumDefinition) keyword defines a C-style enum as a new named type.
+/// Each enum has a fixed number of options which can be specified by name. It is not
+/// possible to set or to example the internal representation of the enum.
+///
+/// ```text
+/// enum exampleNameOfEnum {
+///      /* An identifier for each option, separated by commas */
+///     OPTION_ONE,
+///     OPTION_TWO,
+/// }
+/// ```
+pub(crate) struct EnumDefinition<'src> {
+    name: &'src str,
+    members: Vec<&'src str>,
+}
+
+impl<'src> EnumDefinition<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        map(
+            tuple((
+                kwd("enum"),
+                identifier,
+                sym("{"),
+                comma_separated_list(identifier),
+                sym("}"),
+                sym(";"),
+            )),
+            |(_, name, _, members, _, _)| EnumDefinition { name, members },
+        )(input)
+    }
+}
+
+pub(crate) struct RecordDefinition<'src> {
+    name: &'src str,
+    fields: Vec<FieldDefinition<'src>>,
+}
+
+impl<'src> RecordDefinition<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        map(
+            tuple((
+                kwd("record"),
+                identifier,
+                sym("{"),
+                many0(FieldDefinition::parse_from),
+                sym("}"),
+                sym(";"),
+            )),
+            |(_, name, _, fields, _, _)| RecordDefinition { name, fields },
+        )(input)
+    }
+}
+
+pub(crate) struct FieldDefinition<'src> {
+    name: &'src str,
+    type_: TypeReference<'src>,
+}
+
+impl<'src> FieldDefinition<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        map(
+            tuple((TypeReference::parse_from, identifier, sym(";"))),
+            |(type_, name, _)| FieldDefinition { name, type_ },
+        )(input)
+    }
+}
+
+pub(crate) enum TypeReference<'src> {
+    Boolean,
+    U8,
+    I8,
+    U16,
+    I16,
+    U32,
+    I32,
+    U64,
+    I64,
+    Float,
+    Double,
+    Nullable(Box<TypeReference<'src>>),
+    Sequence(Box<TypeReference<'src>>),
+    Named(&'src str),
+}
+
+impl<'src> TypeReference<'src> {
+    fn parse_from(input: &'src str) -> ParseResult<'src, Self> {
+        let (input, type_) = alt((
+            map(kwd("boolean"), |_| TypeReference::Boolean),
+            map(kwd("u8"), |_| TypeReference::U8),
+            map(kwd("i8"), |_| TypeReference::I8),
+            map(kwd("u16"), |_| TypeReference::U16),
+            map(kwd("i16"), |_| TypeReference::I16),
+            map(kwd("u32"), |_| TypeReference::U32),
+            map(kwd("i32"), |_| TypeReference::I32),
+            map(kwd("u64"), |_| TypeReference::U64),
+            map(kwd("i64"), |_| TypeReference::I64),
+            map(kwd("float"), |_| TypeReference::Float),
+            map(kwd("double"), |_| TypeReference::Double),
+            map(
+                tuple((
```

Comment:
It’s kind of a gratuitous change, but how do you feel about copying Swift’s `[Type]` syntax (or `Type[]`) for sequences, instead of `sequence<Type>`? I could make up a reason about being able to better represent fixed-size arrays as an optimization (`[Type; 12]`), but really it’s because I’ve never been very fond of angle brackets 😆

On the other hand, `sequence<Type>` is nice because it’s consistent with WebIDL...using another syntax would be a divergence for no meaningful reason.